app/src/features/surveys/view/SurveyObservations.tsx

```diff
@@ -79,32 +80,34 @@ const SurveyObservations: React.FC = () => {
 
   const dialogContext = useContext(DialogContext);
 
-  useEffect(() => {
-    const fetchObservationSubmission = async () => {
-      const submission = await biohubApi.observation.getObservationSubmission(projectId, surveyId);
-
-      setSubmissionStatus(() => {
-        setIsLoading(false);
-        if (submission) {
-          if (
-            submission.status === 'Rejected' ||
-            submission.status === 'Darwin Core Validated' ||
-            submission.status === 'Template Validated'
-          ) {
-            setIsValidating(false);
-            setIsPolling(false);
-
-            setPollingTime(null);
-          } else {
-            setIsValidating(true);
-            setIsPolling(true);
-          }
+  const fetchObservationSubmission = useCallback(async () => {
```

Comment:
With this function moved out of the use effect we no longer need to the `pollingFunction` variable, as we can set the polling function in the `useInterval` call from the start (the polling can then just be controlled by the pollingTime variable)

Reply:
This wont really change anything, just a small optimization